src/Filter/TransactionFilter.php

```diff
@@ -15,29 +16,41 @@ class TransactionFilter implements SignableRequest
     /** @var \DateTime */
     private $dateTo;
 
+    /** @var CurrencyCode|null */
+    private $currencyCode;
+
     /**
      * TransactionFilter constructor.
      * @param string $accountIban
      * @param \DateTime $dateFrom
      * @param \DateTime $dateTo
+     * @param string|null $currencyCode
      */
     public function __construct(
         $accountIban,
         \DateTime $dateFrom,
-        \DateTime $dateTo
+        \DateTime $dateTo,
+        $currencyCode = null
     ) {
         $this->accountIban = $accountIban;
         $this->dateFrom = $dateFrom;
         $this->dateTo = $dateTo;
+        $this->currencyCode = $currencyCode !== null ? new CurrencyCode($currencyCode) : null;
     }
 
     public function toArray()
```

Comment:
We could add shaped array annotation for this method, what do U think?

Reply:
we don't use as much shape annotation in this repository, but I don't think it can hurt anything.